### PR TITLE
BH-275 integrated rector refactoring rules 4.0 to 5.0

### PR DESCRIPTION
**Pull request by rfauglas**
**Description (for Contributor and Core Developer)**

PLEASE NOTE CODE UNDER test-src IS NOT PRODUCTIVE CODE IT IS UNTENTIONNALY OLD WAY TO SEE HOW RECTOR BEHAVES.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: std-build/migration/40_to_50/test-src/TestClassRenaming.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<?php
+
+use Akeneo\Connectivity\Connection\Application\Audit\Query\CountDailyEventsByConnectionQuery;
+use Akeneo\Channel\Component\Query\GetChannelCodeWithLocaleCodesInterface;
+
+class TestUsedClassRenaming
+{
+    /** @var CountDailyEventsByConnectionQuery $getErrorCountPerConnectionQuery */
+    private $getErrorCountPerConnectionQuery;
```
Comment (jmleroux) [inline std-build/migration/40_to_50/test-src/TestClassRenaming.php:9]:
```suggestion
    private CountDailyEventsByConnectionQuery $getErrorCountPerConnectionQuery;
```

File: std-build/migration/40_to_50/test-src/TestClassRenaming.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    /**
+     * TestUsedClassRenaming constructor.
+     * @param $getErrorCountPerConnectionQuery
+     */
```
Comment (jmleroux) [inline std-build/migration/40_to_50/test-src/TestClassRenaming.php:14]: PhpDoc is not needed when not usefull.
- the description is just the name
- params are not useful with type hint

File: std-build/migration/40_to_50/test-src/TestClassRenaming.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+// Fail on renaming base class
+//class TestBaseClassRenaming extends CountDailyEventsByConnectionQuery {}
+
+
+class TestUsedInterfaceRenaming
+{
+    /** @var GetChannelCodeWithLocaleCodesInterface $getChannelCodeWithLocaleCodesInterface */
+    private $getChannelCodeWithLocaleCodesInterface;
```
Comment (jmleroux) [inline std-build/migration/40_to_50/test-src/TestClassRenaming.php:28]:
```suggestion
    private GetChannelCodeWithLocaleCodesInterface $getChannelCodeWithLocaleCodesInterface;
```

File: std-build/migration/40_to_50/test-src/TestClassRenaming.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    /**
+     * TestUsedInterfaceRenaming constructor.
+     * @param GetChannelCodeWithLocaleCodesInterface $getChannelCodeWithLocaleCodesInterface
+     */
```
Comment (jmleroux) [inline std-build/migration/40_to_50/test-src/TestClassRenaming.php:33]:
```suggestion
```

File: std-build/migration/40_to_50/test-src/TestClassRenaming.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+
+// Can not rename implemented interface... It fails with Analyze error: "Class Akeneo\Channel\Component\Query\GetChannelCodeWithLocaleCodesInterface not found.
+//class TestClassImplementedInterfaceRenaming implements GetChannelCodeWithLocaleCodesInterface {
+//    public function findAll(): array
+//    {
+//        // TODO: Implement findAll() method.
+//    }
+//}
+
```
Comment (jmleroux) [inline std-build/migration/40_to_50/test-src/TestClassRenaming.php:49]: Please remove dead code before merging

File: std-build/migration/40_to_50/renamed-classes.php [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  'Akeneo\\Pim\\Enrichment\\Component\\Product\\Normalizer\\InternalApi\\JobExecutionNormalizer' => 'Akeneo\\Platform\\Bundle\\ImportExportBundle\\Normalizer\\InternalApi\\JobExecutionNormalizer',
+  'Akeneo\\Tool\\Bundle\\MeasureBundle\\Controller\\ExternalApi\\MeasureFamilyController' => 'Akeneo\\Tool\\Bundle\\MeasureBundle\\Controller\\ExternalApi\\LegacyMeasureFamilyController',
+  'Akeneo\\Tool\\Bundle\\MeasureBundle\\Exception\\UnknownMeasureException' => 'Akeneo\\Tool\\Bundle\\MeasureBundle\\Exception\\MeasurementFamilyNotFoundException',
+  'Akeneo\\Tool\\Bundle\\MeasureBundle\\Exception\\UnknownFamilyMeasureException' => 'Akeneo\\Tool\\Bundle\\MeasureBundle\\Exception\\UnitNotFoundException',
+  'Akeneo\\Pim\\Enrichment\\Component\\Product\\Validator\\Constraints\\WritableDirectory' => 'Akeneo\\Tool\\Component\\StorageUtils\\Validator\\Constraints\\WritableDirectory',
+  'Akeneo\\Pim\\Enrichment\\Component\\Product\\Validator\\Constraints\\WritableDirectoryValidator' => 'Akeneo\\Tool\\Component\\StorageUtils\\Validator\\Constraints\\WritableDirectoryValidator',
+);
```
Comment (mr-prud) [inline std-build/migration/40_to_50/renamed-classes.php:17]:
```suggestion
);

```

**Review by mr-prud — APPROVED** [review]

**Comment by mr-prud** [comment]
check frozen

**Comment by mr-prud** [comment]
hello elsa

**Comment by BitOne** [comment]
Great, good to merge !

**Review by BitOne — APPROVED** [review]

**Pull request merged**